sw/host/opentitanlib/src/io/emu.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::transport::Result;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use thiserror::Error;
+
+/// Error related to the `Emulator` trait. These error messages will be printed in the context of
+/// a TransportError::EmuError, that is "Emulator error: {}".  So including the words "error" or
+/// "Emulator" in texts below will probably be redundant.
+#[derive(Error, Debug, Serialize, Deserialize)]
+pub enum EmuError {
+    #[error("Invalid argument name {0}")]
+    InvalidArgumetName(String),
+    #[error("Argument {0} has invalid value {1}")]
+    InvalidArgumentValue(String, String),
+    #[error("Start failed with cause {0}")]
+    StartFailureCause(String),
+    #[error("Stop failed with cause {0}")]
+    StopFailureCause(String),
+    #[error("Timeout during performing command: {0} timeout: {1}, elsped: {2}")]
+    Timeout(String, u64, u64),
+    #[error("Can't restore resource to initial state: {0}")]
+    ResetError(String),
+    #[error("Runtime error {0}")]
+    RuntimeError(String),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum EmuValue {
+    Empty,
+    String(String),
+    FilePath(String),
+    StringList(Vec<String>),
```

Comment:
Would it also be useful to have a `FilePathList(Vec<String>)` ?  Maybe for a list of apps in some particular order.